app/extend/request.js

```diff
@@ -29,15 +30,18 @@ module.exports = {
    * ```
    */
   get host() {
-    const host = this.get('x-forwarded-host') || this.get('host');
-    if (!host) {
-      return 'localhost';
+    if (this[HOST]) return this[HOST];
+
+    let host;
+    if (this.app.config.proxy) {
```

Comment:
13
down vote
accepted
If you use a front-end service like Apigee as the front-end to your APIs, you will need something like X-FORWARDED-HOST to understand what hostname was used to connect to the API, because Apigee gets configured with whatever your backend DNS is, nginx and your app stack only see the Host header as your backend DNS name, not the hostname that was called in the first place.